lib/Doctrine/DBAL/Types/ConversionException.php

```diff
@@ -36,14 +36,19 @@ class ConversionException extends \Doctrine\DBAL\DBALException
      *
      * @param string $value
      * @param string $toType
+     * @param \Exception|null $previous
      *
      * @return \Doctrine\DBAL\Types\ConversionException
      */
-    static public function conversionFailed($value, $toType)
+    static public function conversionFailed($value, $toType, \Exception $previous = null)
```

Comment:
@greg0ire subclassing and implementing there then? `throw CouldNotCreatePSR7Uri::fromUriException($dbValue, $exception)`, where `CouldNotCreatePSR7Uri` extends `ConversionException`